drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1146 @@
+	if (ret)
+		return ret;
+
+	/* 2. Clear the LOW_POWER bit (Bit D4) in the BW_RATE register. */
+	ret = adxl314_set_low_power_mode(dev, DISABLE_E);
+	if (ret)
+		return ret;
+
+	/* 3. Read samples from FIFO. (Recommended starting point is 0.1 sec worth
+	 * of data for data rates of 100 Hz or greater).  */
+
+	/* Set FIFO mode to FIFO */
+	ret = adxl314_set_fifo_mode(dev, ADXL314_FIFO_MODE);
+	if (ret)
+		return ret;
+
+	/* Set dev. in measure mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_MEAS);
+	if (ret)
+		return ret;
+
+	/* Output settling time, min 4 samples. 10ms per sample at 100Hz. */
+	no_os_mdelay(50);
+
+	/* 4. Read and store the 10 samples in the FIFO. */
+	ret = adxl314_get_raw_fifo_data(dev, &fifo_entries, x, y, z);
+	if (ret)
+		return ret;
+
+	/* Compute average (10 samples). */
+	for (uint8_t idx = 0; idx < ADXL314_ST_SAMPLES; idx++) {
+		zst_off += z[idx];
+	}
+	zst_off /= ADXL314_ST_SAMPLES;
+
+	/* 5. Enable self-test. */
+	/* Set dev in standby mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_STDBY);
+	if (ret)
+		return ret;
+
+	/* Set self-test bit. */
+	ret = adxl314_control_reg_bit(dev, ADXL314_REG_DATA_FORMAT,
+				      ENABLE_E, ADXL314_REG_DATA_FORMAT_SELF_TEST);
+	if (ret)
+		return ret;
+
+	/* Set dev. in measure mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_MEAS);
+	if (ret)
+		return ret;
+
+	/* Output settling time, min 4 samples. 10ms per sample at 100Hz. */
+	no_os_mdelay(50);
+
+	/* 6. Read and store samples in the FIFO. */
+	ret = adxl314_get_raw_fifo_data(dev, &fifo_entries, x, y, z);
+	if (ret)
+		return ret;
+
+	/* 7. Compute average (10 samples). */
+	for (uint8_t idx = 0; idx < ADXL314_ST_SAMPLES; idx++) {
+		zst_on += z[idx];
+	}
+	zst_on /= ADXL314_ST_SAMPLES;
+
+	/* 8. Disable self-test. */
+	/* Set dev. in standby mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_STDBY);
+	if (ret)
+		return ret;
+
+	/* Reset self-test bit. */
+	ret = adxl314_control_reg_bit(dev, ADXL314_REG_DATA_FORMAT,
+				      DISABLE_E, ADXL314_REG_DATA_FORMAT_SELF_TEST);
+	if (ret)
+		return ret;
+
+	/* Set dev. in measure mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_MEAS);
+	if (ret)
+		return ret;
+
+	zst = zst_on - zst_off;
+
+	if ((zst * ADXL314_ST_MULT) < ADXL314_ST_TYP_DEVIATION)
+		pr_info("Self-test passed.\n");
+	else
+		pr_info("Self-test failed.\n");
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Convert raw acceleration value to g value.
+ *
+ * @param raw_accel - Raw acceleration value.
+ *
+ * @return ret      - Converted data.
+*******************************************************************************/
+static int64_t adxl314_accel_conv(uint16_t raw_accel)
+{
+	int64_t accel_data;
+
+	/* Convert from 16-bit unsigned to 16-bit signed number. */
+	accel_data = (int16_t)raw_accel;
+
+	/* Apply scale factor based on the selected range. */
+	return ((int64_t)(accel_data * ADXL314_ACC_SCALE_FACTOR_MUL));
```

Comment:
is this cast really required? (accel_data is already int64_t)

Reply:
You are right, there is no need for the cast here.